Mathlib/Data/Nat/Fib/Zeckendorf.lean

```diff
@@ -0,0 +1,175 @@
+/-
+Copyright (c) 2023 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Data.Nat.Fib.Basic
+
+/-!
+# Zeckendorf's Theorem
+
+This file proves Zeckendorf's theorem: Every natural number can be written uniquely as a sum of
+distinct non-consecutive Fibonacci numbers.
+
+## Main declarations
+
+* `List.IsZeckendorfRep`: Predicate for a list to be an increasing sequence of non-consecutive
+  natural numbers greater than or equal to `2`, namely a Zeckendorf representation.
+* `Nat.greatestFib`: Index of the greatest Fibonacci number less than some natural number.
+* `Nat.zeckendorf`: Send a natural number to its Zeckendorf representation.
+* `Nat.zeckendorfEquiv`: Zeckendorf's theorem, in the form of an equivalence between natural numbers
+  and Zeckendorf representations.
+
+## TODO
+
+We could prove that the order induced by `zeckendorfEquiv` on Zeckendorf representations is exactly
+the lexicographic order.
+
+## Tags
+
+fibonacci, zeckendorf, digit
+-/
+
+open List Nat
+open scoped BigOperators
+
+local instance : IsTrans ℕ λ a b ↦ b + 2 ≤ a where
```

Comment:
```suggestion
local instance : IsTrans ℕ fun a b ↦ b + 2 ≤ a where
```
and throughout

Reply:
Do we have any style guide regarding this? I don't think the community as a whole has come to the conclusion that we should avoid `λ`.

Reply:
We systematically eliminated every use of `λ` with mathport. Any that remain are almost certainly due to manual porting.

Reply:
Okay, so this is a [mathportism](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/mathlib4.3A.20autoImplicit.20false.20by.20default.3F/near/384422977), not a style recommendation.

Reply:
Just because it was implemented by mathport doesn't make it a mathportism. This reflects Lean 4 style, and I think it's been mentioned other places that we prefer to avoid `λ` for anonymous functions.

Reply:
This is very vague but because you ask nicely. I'll stick to `fun` *for now*.